specification/0002-quilt.mod.json.md

```diff
@@ -396,11 +396,34 @@ A dependency object defines what mods/plugins a given mod depends on or breaks.
 A mod identifier in the form of either `mavenGroup:modId` or `modId`.
 
 ### The `versions` field
-| Type         | Required | Default |
-|--------------|----------|---------|
-| Array/String | False    | `"*"`   |
+| Type                | Required | Default |
+|---------------------|----------|---------|
+| Array/Object/String | False    | `"*"`   |
 
-Should be a [version specifier](#version-specifier) or array of version specifiers defining what versions this dependency applies to. If an array of versions is provided, the dependency matches if it matches ANY of the listed versions.
+A version specifier, or complex set of version specifiers that control what versions match this dependency object.
+
+It is an error to specify multiple constraints that conflict with each other (where no version would match the whole specifier).
+
+It is an error if the resulting version specifier matches every version, and isn't the single string `"*"`, or uses the deprecated array style of definition.
```

Comment:
You probably want `when` here, not `if`.

Reply:
Yes, thanks!